dali/kernels/signal/window/extract_windows_gpu.cuh

```diff
@@ -501,34 +501,38 @@ struct ExtractHorizontalWindowsGpuImpl : ExtractWindowsGpuImpl<Dst, Src> {
 
     constexpr int kDefaultBlockSize = 256;
     logical_block_size = kDefaultBlockSize;
-    int64_t max_len = 0;
+    int64_t max_padded_length = 0;
     int max_win_per_input = 0;
     for (int i = 0; i < N; i++) {
-      int64_t length = lengths[i][0];
-      if (length > max_len)
-        max_len = length;
+      int64_t length = lengths[i];
       int nwin = args.num_windows(length);
+
+      int64_t padded_length = static_cast<int64_t>(nwin-1) * args.window_step + args.window_length;
+      if (padded_length > max_padded_length)
+        max_padded_length = padded_length;
```

Comment:
This is a bug fix.

Reply:
Can you add some test case to covert this fixes?

Reply:
It's there. extract_windows_gpu_test.cu:219